File: leetcode/2medium/이경호/3376.cpp
class Solution {
public:
  int findMinimumTime(vector<int>& strengths, int k) {
    int ans = 10000000;
    sort(strengths.begin(), strengths.end());
    do {
      int t = 0;
      int x = 1;
      for (int strength : strengths) {
        t += strength / x;
        if (strength % x) {
          t++;
        }
        x += k;
      }
      ans = std::min(ans, t);
    } while (next_permutation(strengths.begin(), strengths.end()));
    return ans;
  }
};

File: leetcode/2medium/위성원/3376.ts
function findMinimumTime(strength: number[], k: number): number {
  let x = 1;
  let time = 0;
  let energy = 0;
  let s = [...strength].sort((a, b) => b - a);

  while (s.length > 0) {

    time++;
    energy += x;

    for (let i = 0; i < s.length; i++) {
      if (energy >= s[i]) {
        energy = 0;
        x += k;
        s = [...s.slice(0, i), ...s.slice(i + 1, s.length)]
        break;
      }
    }
  }

  return time
};